### PR TITLE
Fix `Parameter.is_real()`

### LINKED ISSUE #8619: Parameter is_real seems not correct
### Environment

- **Qiskit Terra version**: 
- **Python version**: 
- **Operating system**: 

### What is happening?

`Parameter.is_real()` seems not correct.

### How can we reproduce the issue?

```python
>>> from qiskit.circuit import Parameter

>>> Parameter("a").is_real()
True
>>> (1j * Parameter("a")).is_real()
True
```

### What should happen?

Both should return `False` since we don't know whether `a` is real or not.

### Any suggestions?

_No response_

### PR DESCRIPTION
**Pull request by king-p3nguin**
### Summary
- Resolve #8619 
- Add test for `Parameter.is_real()`

### CODE REVIEW

**Comment by Cryoris** [comment]
This LGTM thanks for the fix! @mtreinish does the CI currently only test with either SymPy or SymEngine, or is there a way to ensure both work?

**Comment by jakelishman** [comment]
Since `symengine` is an optional, you can use the `optionals.HAS_SYMENGINE.disable_locally` context manager to cause the lazy checkers to return `False` within the body of the `with`.  If you construct the `Parameter` objects and manipulate them within that, it'll be forced to use `sympy` constructs.

**Comment by mtreinish** [comment]
Besides @jakelishman's suggestion which is what you should do to test both libraries explicitly. We have implicit coverage in CI because the windows jobs do not run with symengine installed https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=44381&view=logs&j=2ade34fd-5d97-54e5-ba94-6bcbab8d6365&t=ef2b524a-d9fe-5838-e305-d18c09dc7083&l=42 because of a limitation with python packaging's ability to specify a requirement is only applicable on 64bit windows. (I have an issue open about this here: https://github.com/pypa/interoperability-peps/issues/74 )

**Comment by Cryoris** [comment]
If it's implicitly tested I think it's fine -- we should add proper symengine + sympy tests with @jakelishman's suggestion but that's probably a separate PR 🙂

File: releasenotes/notes/fix-parameter-is_real-8b8f99811e58075e.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Fixed a bug where :meth:`.Parameter.is_real` did not return ``None`` when
+    the parameter is not bound.
+    Fixed `#8619 <https://github.com/Qiskit/qiskit-terra/issues/8619>`.
```
Comment (Cryoris) [inline releasenotes/notes/fix-parameter-is_real-8b8f99811e58075e.yaml:6]:
```suggestion
    Fixed `#8619 <https://github.com/Qiskit/qiskit-terra/issues/8619>`__.
```

**Review by Cryoris — APPROVED** [review]

**Comment by Cryoris** [comment]
I didn't tag this as backport as this might be an unexpected change for a bugfix release (even though we labelled it bugfix).

**Comment by jakelishman** [comment]
Yeah agreed, we're probably ok to actually change the return value (especially since it should fail safe as `bool(None) is False`), but that's a bit much for a backport.

**Pull request merged**